pkg/southbound/topocache/topocache.go

```diff
@@ -22,114 +22,69 @@ Until then this simple cache will load a set of Device definitions from file
 package topocache
 
 import (
-	"encoding/json"
-	"fmt"
+	"context"
 	"github.com/onosproject/onos-config/pkg/events"
-	"os"
-	"regexp"
+	deviceproto "github.com/onosproject/onos-topo/pkg/northbound/proto"
+	"google.golang.org/grpc"
+	"io"
+	log "k8s.io/klog"
 )
 
-const storeTypeDevice = "device"
-const storeVersion = "1.0.0"
-const deviceIDNamePattern = `[a-zA-Z0-9\-:_]{4,40}`
-const deviceVersionPattern = `[a-zA-Z0-9_\.]{2,10}` //same as configuration.go
-
-// ID is an alias for string
-type ID string
-
-// Device - the definition of Device will ultimately come from onos-topology
-type Device struct {
-	ID                                                ID
-	Addr, Target, Usr, Pwd, CaPath, CertPath, KeyPath string
-	Plain, Insecure                                   bool
-	Timeout                                           int64
-	SoftwareVersion                                   string
-}
+const (
+	topoAddress = "onos-topo:5150"
+)
 
 // DeviceStore is the model of the Device store
 type DeviceStore struct {
-	Version     string
-	Storetype   string
-	Store       map[ID]Device
-	topoChannel chan<- events.TopoEvent
+	client deviceproto.DeviceServiceClient
 }
 
-// LoadDeviceStore loads a device store from a file - will eventually be from onos-topology
-func LoadDeviceStore(file string, topoChannel chan<- events.TopoEvent) (*DeviceStore, error) {
-	storeFile, err := os.Open(file)
+// LoadDeviceStore loads a device store
+func LoadDeviceStore(topoChannel chan<- events.TopoEvent, opts ...grpc.DialOption) (*DeviceStore, error) {
```

Comment:
maybe this could be renamed to CacheDeviceStore - it's basically a GET on onos-topo I'm guessing